pkg/controller/migration/convert/cni.go

```diff
@@ -1,14 +1,47 @@
 package convert
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
+	"io"
 	"log"
+	"net"
+	"reflect"
 	"strings"
 
 	"github.com/containernetworking/cni/libcni"
+	"github.com/containernetworking/cni/pkg/types"
 )
 
+// Used to parse the IPAM section out of the full CNI configuration
+type CNIHostLocalIPAM struct {
+	IPAM json.RawMessage `json:"ipam"`
+}
+
+// IPAMConfig represents the IP related network configuration.
+// This nests Range because we initially only supported a single
+// range directly, and wish to preserve backwards compatability
+type HostLocalIPAMConfig struct {
+	*Range
+	Name       string
+	Type       string         `json:"type"`
+	Routes     []*types.Route `json:"routes"`
+	DataDir    string         `json:"dataDir"`
+	ResolvConf string         `json:"resolvConf"`
```

Comment:
Do we need _all_ of this stuff?

Reply:
Not really, it is just what I copied over from containernetworking/plugins

Reply:
Can we reproduce their config if they've set `ResolvConf`? I think you need to drop any field in here that we don't know how to migrate so that the later unmarshal using `DisallowUnknownFields` will raise an error